### PR TITLE
porting pblintd function

### PR DESCRIPTION
**Pull request by xyuan**
porting pblintd function

[BFB]

### CODE REVIEW

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 1]
```diff
@@ -3188,9 +3188,77 @@ void pblintd_check_pblh_f(Int shcol, Int nlev, Int nlevi, Int npbl, Real* z, Rea
...
+
+  std::vector<view_1d> views_1d(3);
+  ekat::host_to_device({ustar, obklen, kbfs}, shcol, views_1d);
+  view_1d ustar_1d   (views_1d[0]),
+          obklen_1d  (views_1d[1]),
+          kbfs_1d    (views_1d[2]);
+
+  SHOC::view_1d<bool> check_1d("check", shcol);
+  const auto host_check_1d = Kokkos::create_mirror_view(check_1d);
+  Kokkos::deep_copy(check_1d, host_check_1d);
```
Comment (tcclevenger) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:3223]: Use `ekat::host_to_device` for bools

Comment (xyuan) [reply components/scream/src/physics/shoc/shoc_functions_f90.cpp:3223]: sorry, it is a local variable, so we don't need this one, I deleted it though.

**Comment by xyuan** [comment]
@tcclevenger could you please help me to manually test this PR on blake? I tested it on Nersc and ANL machine using the same compiler but different version of OpenMPI, they are all passed the test. It is curious for me to understand what is the reason for this PR to be failed on black and mappy.

**Comment by tcclevenger** [comment]
> @tcclevenger could you please help me to manually test this PR on blake? I tested it on Nersc and ANL machine using the same compiler but different version of OpenMPI, they are all passed the test. It is curious for me to understand what is the reason for this PR to be failed on black and mappy.

@xyuan Yeah, I'll take a look

**Comment by tcclevenger** [comment]
@xyuan I believe I found the issue. I'll either push to this branch or create a new PR with the solution.

**Comment by xyuan** [comment]
Alright, thanks very much

**Comment by tcclevenger** [comment]
@xyuan This function is now ported in https://github.com/E3SM-Project/scream/pull/853. I tried for a while to find the issue with the tests here, but was unable. But the tests are passing in the other PR.

**Comment by xyuan** [comment]
close thie PR

**Pull request closed without merging**